crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -359,3 +385,222 @@ impl ProgrammableTransactionBuilder {
+
+        let arg = builder.smash_coins(
+            vec![target_coin_ref]
+        ).unwrap();
+
+        let tx = builder.finish();
+
+        assert_eq!(arg, Input(0));
+        assert_eq!(
+            tx.inputs,
+            vec![CallArg::Object(ObjectArg::ImmOrOwnedObject(target_coin_ref))]
+        );
+        assert_eq!(
+            tx.commands,
+            vec![
+                Command::MergeCoins(
+                    Input(0),
+                    vec![]
+                )
+            ]
+        );
+    }
+
+    #[test]
+    fn test_builder_smash_coins_two_coins() {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        let target_coin_ref = random_object_ref();
+        let source_coin_ref = random_object_ref();
+
+        let arg = builder.smash_coins(
+            vec![
+                target_coin_ref,
+                source_coin_ref
+            ]
+        ).unwrap();
+
+        let tx = builder.finish();
+
+        assert_eq!(arg, Input(0));
+        assert_eq!(
+            tx.inputs,
+            vec![
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(target_coin_ref)),
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(source_coin_ref))
+            ]
+        );
+        assert_eq!(
+            tx.commands,
+            vec![
+                Command::MergeCoins(
+                    Input(0),
+                    vec![Input(1)]
+                )
+            ]
+        );
+    }
+
+    #[test]
+    fn test_builder_smash_coins_three_coin() {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        let target_coin_ref = random_object_ref();
+        let source_coin1_ref = random_object_ref();
+        let source_coin2_ref = random_object_ref();
+
+        let arg = builder.smash_coins(
+            vec![
+                target_coin_ref,
+                source_coin1_ref,
+                source_coin2_ref,
+            ]
+        ).unwrap();
+
+        let tx = builder.finish();
+
+        assert_eq!(arg, Input(0));
+        assert_eq!(
+            tx.inputs,
+            vec![
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(target_coin_ref)),
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(source_coin1_ref)),
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(source_coin2_ref))
+            ]
+        );
+        assert_eq!(
+            tx.commands,
+            vec![
+                Command::MergeCoins(
+                    Input(0),
+                    vec![
+                        Input(1),
+                        Input(2),
+                    ]
+                )
+            ]
+        );
+    }
+
+    #[test]
+    fn test_builder_smash_coins_zero_coin() {
+        let mut builder = ProgrammableTransactionBuilder::new();
+
+        let err = builder.smash_coins(
+            vec![]
+        ).err().unwrap();
+
+        assert_eq!(err.to_string(), "coins vector is empty");
```

Comment:
I don't think the error message here is that important, and we can simply do `assert!(builder.smash_coins(vec![]).is_err())`

Reply:
done!